### PR TITLE
Rename to ubi7/8 images. Remove jdk8/11 images

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -13,14 +13,14 @@ ENV \
     NSS_WRAPPER_PASSWD="/etc/passwd" \
     CLI_JAR_DOWNLOAD_SHA256=$CLI_JAR_SHA \
     ANALYZER_INIT_DOWNLOAD_SHA256=$ANALYZER_INIT_SHA \
-    PATH="$JAVA_HOME/bin:$PATH" \
     JAVA_HOME="/usr/lib/jvm/java-$JAVA_PACKAGE" \
     JAVA_VENDOR="openjdk" \
-    JAVA_VERSION="$JAVA_PACKAGE"
+    JAVA_VERSION="$JAVA_PACKAGE" \
```
Comment (stuartwdouglas) [inline Dockerfile.template]: I don't think we need this line any more.

File: Dockerfile.template [hunk 2]
```diff
@@ -55,7 +55,7 @@ RUN set -o errexit -o nounset \
     && chmod 755 ${GRADLE_MANIPULATOR_HOME}/bin/gradle-manipulator \
     && ln --symbolic "${GRADLE_MANIPULATOR_HOME}/bin/gradle-manipulator" /usr/bin/gradle-manipulator
     #we test out gradle manipulator as part of the gradle installation
-
+# YYY
```
Comment (stuartwdouglas) [inline Dockerfile.template]: Is this something left over?

File: generate.sh [hunk 3]
```diff
@@ -53,9 +53,15 @@ generate () {
...
 
-  envsubst '$IMAGE_NAME,$BASE_IMAGE,$MAVEN_VERSION,$MAVEN_SHA,$GRADLE_VERSION,$GRADLE_SHA,$GRADLE_MANIPULATOR_VERSION,$CLI_JAR_SHA,$ANALYZER_INIT_SHA,$TOOL_STRING,$JAVA_PACKAGE' < $DIR/Dockerfile.template > $DIR/$IMAGE_NAME/Dockerfile
+  envsubst '$IMAGE_NAME,$BASE_IMAGE,$MAVEN_VERSION,$MAVEN_SHA,$GRADLE_VERSION,$GRADLE_SHA,$GRADLE_MANIPULATOR_VERSION,$CLI_JAR_SHA,$ANALYZER_INIT_SHA,$TOOL_STRING,$JAVA_PACKAGE,$UBI' < $DIR/Dockerfile.template > $DIR/$IMAGE_NAME/Dockerfile
   envsubst '$IMAGE_NAME,$BASE_IMAGE' < $DIR/push.yaml > $DIR/.tekton/$IMAGE_NAME-push.yaml
   envsubst '$IMAGE_NAME,$BASE_IMAGE' < $DIR/pull-request.yaml > $DIR/.tekton/$IMAGE_NAME-pull-request.yaml
+
+  if [ "$UBI" == "7" ]; then
+      sed -i "/# XXX/,/YYY/d" $DIR/$IMAGE_NAME/Dockerfile
+      sed -i "s/--setopt=install_weak_deps=0 //" $DIR/$IMAGE_NAME/Dockerfile
+      sed -i "s/buildah bzip2-devel cmake emacs-filesystem glibc-langpack-en glibc-static golang java-17-openjdk-devel libstdc++-static podman/java-1.7.0-openjdk-devel/g" $DIR/$IMAGE_NAME/Dockerfile
```
Comment (stuartwdouglas) [inline generate.sh]: I don't really like this approach. I would rather have 2 explicit package lists of the 'extra' stuff that goes into both images, and replace them in a template variable.

File: image-config.yaml [hunk 4]
```diff
@@ -7,7 +7,7 @@ spec:
   #Versions places here for gradle will be used generate the relevant dockerfiles
   #after you run generate.sh
   builders:
-    jdk7:
-      tag: jdk:7,maven:3.8,ant:1.9.16
-    jdk17:
+    ubi7:
+      tag: jdk:7,maven:3.8.8,ant:1.9.16
```
Comment (stuartwdouglas) [inline image-config.yaml]: I have had to push this change separately, as the JDK7 build was broken without it.

File: generate.sh [hunk 5]
```diff
@@ -53,9 +53,15 @@ generate () {
 
   export TOOL_STRING="$TOOL_STRING"
 
-  envsubst '$IMAGE_NAME,$BASE_IMAGE,$MAVEN_VERSION,$MAVEN_SHA,$GRADLE_VERSION,$GRADLE_SHA,$GRADLE_MANIPULATOR_VERSION,$CLI_JAR_SHA,$ANALYZER_INIT_SHA,$TOOL_STRING,$JAVA_PACKAGE' < $DIR/Dockerfile.template > $DIR/$IMAGE_NAME/Dockerfile
+  envsubst '$IMAGE_NAME,$BASE_IMAGE,$MAVEN_VERSION,$MAVEN_SHA,$GRADLE_VERSION,$GRADLE_SHA,$GRADLE_MANIPULATOR_VERSION,$CLI_JAR_SHA,$ANALYZER_INIT_SHA,$TOOL_STRING,$JAVA_PACKAGE,$UBI' < $DIR/Dockerfile.template > $DIR/$IMAGE_NAME/Dockerfile
   envsubst '$IMAGE_NAME,$BASE_IMAGE' < $DIR/push.yaml > $DIR/.tekton/$IMAGE_NAME-push.yaml
   envsubst '$IMAGE_NAME,$BASE_IMAGE' < $DIR/pull-request.yaml > $DIR/.tekton/$IMAGE_NAME-pull-request.yaml
+
+  if [ "$UBI" == "7" ]; then
+      sed -i "/# XXX/,/YYY/d" $DIR/$IMAGE_NAME/Dockerfile
```
Comment (stuartwdouglas) [inline generate.sh]: What is this?

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

**Comment by stuartwdouglas** [comment]
Now we are using the new style images it would be good to get this in.

File: generate.sh (hunk 3 above)
Comment (rnc) [reply generate.sh]: Agreed, and done.

File: generate.sh (hunk 5 above)
Comment (rnc) [reply generate.sh]: An experimental initial attempt, since removed, to use a singlular docker template.

**Comment by rnc** [comment]
> Now we are using the new style images it would be good to get this in.

While I've pushed an update given this will have some side affects (e.g. dockerfile removal) I think it needs discussion. I'm not sure how the `.tekton'` directories and being used and what we'll need to change in JBS itself. And potentially the entire hacbs-jdk7-builder / hacbs-jdk17-builder could be renamed to hacbs-ubi7-builder / hacbs-ubi8-builder ?

I think the only references in JBS are in 
```
pkg/reconciler/dependencybuild/dependencybuild_test.go 
java-components/management-console/src/test/resources/dependencybuild/quarkus-http-2023-10-30.yaml 
deploy/operator/config/system-config.yaml
```

**Comment by stuartwdouglas** [comment]
We probably need a solution for digest pinning, but lets get this in for now.

**Pull request merged**